### PR TITLE
Add crux-transpose-windows for swapping the buffers between two windows

### PR DESCRIPTION
**Pull request by dgtized**
This is copied directly from
https://www.emacswiki.org/emacs/TransposeWindows, and renamed to
crux-tranpose-windows. The wiki page mentions the original author as
Thomas Bellman. It does not appear to be published in any existing
package that I could find.

### CODE REVIEW

**Comment by bbatsov** [comment]
Looks good, just squash the two commits together. Adding a brief description of the difference with `swap-windows` would be useful as well I guess.

**Comment by dgtized** [comment]
Squashed the commits together. In terms of the difference between transpose and swap, swap only works for 2 open windows (and only for two, it errors out if there are more). To be honest, I'm not sure I can think of any case where swap-windows would be preferable to transpose. As far as I know, transpose is a superset of the functionality of swap-windows.

File: crux.el [hunk 1]
```diff
@@ -346,6 +346,22 @@ buffer is not visiting a file."
...
+transpositions to execute in sequence."
+  (interactive "p")
+  (let ((selector (if (>= arg 0) 'next-window 'previous-window)))
+    (while (/= arg 0)
+      (let ((this-win (window-buffer))
+            (next-win (window-buffer (funcall selector))))
+        (set-window-buffer (selected-window) next-win)
+        (set-window-buffer (funcall selector) this-win)
+        (select-window (funcall selector)))
+      (setq arg (if (plusp arg) (1- arg) (1+ arg))))))
```
Comment (bbatsov) [inline crux.el]: Btw, this should be `cl-plusp`. (or just `(< 0 arg)`)

**Comment by bbatsov** [comment]
> As far as I know, transpose is a superset of the functionality of swap-windows.

Probably we can mark the other function as obsolete and remove it down the road then.

**Comment by dgtized** [comment]
Fixed the cl-plusp issue, and documented the behavior on prefix arguments like 2 or -2.

The only part about swap-windows that I don't follow is why it calls `set-window-start`, I think that is actually unnecessary but not sure. I'm fine with marking it deprecated. I think you could even alias it to transpose windows, and no one would notice.

**Comment by bbatsov** [comment]
> I think you could even alias it to transpose windows, and no one would notice.

Good idea. Alias it and remove it from the readme.

**Comment by dgtized** [comment]
Should that also be squashed in, a second commit or a different PR?

**Comment by bbatsov** [comment]
Second commit, same PR.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**